src/Arcus.Messaging.ServiceBus.Core/Extensions/AzureClientFactoryBuilderExtensions.cs

```diff
@@ -61,19 +65,38 @@ public static class AzureClientFactoryBuilderExtensions
             
             return builder.AddClient<ServiceBusClient, ServiceBusClientOptions>((options, serviceProvider) =>
             {
-                var secretProvider = serviceProvider.GetService<ISecretProvider>();
-                if (secretProvider is null)
-                {
-                    throw new InvalidOperationException(
-                        "Requires an Arcus secret store registration to retrieve the connection string to authenticate with Azure Service Bus while creating an Service Bus client instance," 
-                        + "please use the 'services.AddSecretStore(...)' or 'host.ConfigureSecretStore(...)' (https://security.arcus-azure.net/features/secret-store)");
-                }
-
-                string connectionString = secretProvider.GetRawSecretAsync(connectionStringSecretName).GetAwaiter().GetResult();
+                string connectionString = GetServiceBusConnectionString(connectionStringSecretName, serviceProvider);
                 configureOptions?.Invoke(options);
 
                 return new ServiceBusClient(connectionString, options);
             });
         }
+
+        private static string GetServiceBusConnectionString(string connectionStringSecretName, IServiceProvider serviceProvider)
+        {
+            var secretProvider = serviceProvider.GetService<ISecretProvider>();
+            if (secretProvider is null)
+            {
+                throw new InvalidOperationException(
+                    "Requires an Arcus secret store registration to retrieve the connection string to authenticate with Azure Service Bus while creating an Service Bus client instance," 
+                    + "please use the 'services.AddSecretStore(...)' or 'host.ConfigureSecretStore(...)' (https://security.arcus-azure.net/features/secret-store)");
+            }
+
+            try
+            {
+                string connectionString = secretProvider.GetRawSecret(connectionStringSecretName);
+                return connectionString;
+            }
+            catch (Exception exception)
+            {
+                ILogger logger = 
+                    serviceProvider.GetService<ILogger<ServiceBusClient>>() 
+                    ?? NullLogger<ServiceBusClient>.Instance;
+
+                logger.LogWarning(exception, "Cannot synchronously retrieve Azure Service Bus connection string secret for '{SecretName}', fallback on asynchronously", connectionStringSecretName);
```

Comment:
Why is this a warning ?  I would categorize this more as a trace.

Reply:
Yes, updated to `LogTrace` in https://github.com/arcus-azure/arcus.messaging/pull/373/commits/efbcb36c397809e914f3951916188d0c51fc8344